AR-js-org/AR.js
# Custom arjs-look-controls component and smoothing to reduce shaking effect

**⚠️ All PRs have to be done versus 'dev' branch, so be aware of that, or we'll close your issue ⚠️**

**What kind of change does this PR introduce?**

Feature; considerably reduces the shaking effect when holding device still. 

**Can it be referenced to an Issue? If so what is the issue # ?**
#139 

**How can we test it?**


Note that you cannot use the `<a-camera>` tag as this automatically injects the default `look-controls`. So you hace to create an `<a-entity>` with a `camera` component, as well as `arjs-look-controls`.

So for example:
`<a-entity id='camera1' camera='near:0.1; far:50000; fov:80;' arjs-look-controls='smoothingFactor: 0.1' gps-projected-camera='gpsMinDistance: 5' ... >`
**Summary**


Considerably reduces the shaking effect when holding device still. It does this by applying smoothing to the raw orientation values.

To do this, I created a custom `look-controls` component entitled `arjs-look-controls` to replace the default one. This approach is recommended in the A-Frame documentation, see [here](https://aframe.io/docs/1.0.0/components/look-controls.html). The component uses `THREE.DeviceOrientationControls` which itself has been modified to add smoothing. The `arjs-look-controls` component has a `smoothingFactor` property to control the amount of smoothing.

A smoothed orientation is essentially defined by:
`smoothedOrientation  = smoothingFactor * currentValue + (1-smoothingFactor) * previousSmoothedValue`

so it can be seen that the lower the smoothing factor (in range 0 to 1), the more smoothing applies. In tests 0.1 seems to work well whereas 0.05 smooths a little too much.

**Does this PR introduce a breaking change?**

No. Code to test for `arjs-look-controls` has been added to `gps-camera` and `gps-projected-camera` but it falls back to `look-controls` if not available.

**Please TEST your PR before proposing it. Specify here what device you have used for tests, version of OS and version of Browser**

Pixel 3, Android 10, Chrome.

**Other information**
